### PR TITLE
chore: small improvements

### PR DESCRIPTION
**Pull request by mds1**
- disable slither reentrancy detector in scripts
- pin solc to 0.8.20
- only use absolute imports

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -1,16 +1,16 @@
 // SPDX-License-Identifier: UNLICENSED
+// slither-disable-start reentrancy-benign
 
-pragma solidity ^0.8.16;
+pragma solidity 0.8.20;
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply script/Deploy.s.sol:4]: This seems right to me, and doesn't 0.8.20 have issues with L2 compatibility?

**Review by jferas — APPROVED** [review]
LGTM

File: script/Deploy.s.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (mds1) [reply script/Deploy.s.sol:4]: solc will only generate the PUSH0 opcode in bytecode if the `evm_version` is set to Shanghai

solc and hardhat both use Shanghai as the default, but foundry ended up sticking with paris as the default to help avoid the footgun you called out (bytecode with PUSH0 that can't be deployed on many chains)

But it's good point, so I just pushed dc295ac to explicitly specify the EVM version

File: src/Counter.sol [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.16;
+pragma solidity 0.8.20;
```
[the remark this answers is not in the document]
Comment (mds1) [reply src/Counter.sol:2]: Updated CI in aa907c9 (and docs in 51f23f7) to ignore the version check by default

**Review by alexkeating — APPROVED** [review]

**Pull request merged**